src/main/java/io/pravega/connectors/hadoop/PravegaClientConfig.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.connectors.hadoop;
+
+import com.google.common.base.Preconditions;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.impl.Credentials;
+import lombok.Data;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Helper class to assist building Pravega client configurations.
+ */
+public class PravegaClientConfig {
+
+    static final PravegaParameter CONTROLLER_PARAM = new PravegaParameter("controller", "pravega.controller.uri", "PRAVEGA_CONTROLLER_URI");
+    static final PravegaParameter SCOPE_PARAM = new PravegaParameter("scope", "pravega.scope", "PRAVEGA_SCOPE");
+
+    private static final long serialVersionUID = 1L;
+
+    private URI controllerURI;
+    private String defaultScope;
+    private Credentials credentials;
+    private boolean validateHostname = true;
+    private String trustStore;
+
+    PravegaClientConfig(Properties properties, Map<String, String> env, Map<String, String> params) {
+        Preconditions.checkNotNull(properties, "Properties cannot be null");
+        Preconditions.checkNotNull(env, "env cannot be null");
+        Preconditions.checkNotNull(params, "params cannot be null");
+        this.controllerURI = CONTROLLER_PARAM.resolve(params, properties, env).map(URI::create).orElse(null);
+        this.defaultScope = SCOPE_PARAM.resolve(params, properties, env).orElse(null);
+    }
+
+    /**
+     * Gets a configuration based on defaults obtained from the local environment.
+     */
+    public static PravegaClientConfig fromDefaults() {
+        return new PravegaClientConfig(System.getProperties(), System.getenv(), new HashMap<>());
+    }
+
+    /**
+     * Gets a configuration based on defaults obtained from the local environment plus the given program parameters.
+     *
+     * @param params the parameters to use.
+     */
+    public static PravegaClientConfig fromParams(Map<String, String> params) {
+        return new PravegaClientConfig(System.getProperties(), System.getenv(), params);
+    }
+
+    /**
+     * Gets the {@link ClientConfig} to use with the Pravega client.
+     */
+    public ClientConfig getClientConfig() {
+        ClientConfig.ClientConfigBuilder builder = ClientConfig.builder()
+                .validateHostName(validateHostname);
```

Comment:
validateHostName is only applicable when TLS is enabled.

Reply:
The helper class does not check if security is enabled or not. It simply passes the configuration and by default the value is false.The validate hostname property will not be used on the Pravega side if TLS is not enabled.